Sources/Amplitude/Utilities/HttpClient.swift

```diff
@@ -37,10 +37,19 @@ class HttpClient {
             let request = try getRequest()
             let requestData = getRequestData(events: events)
 
-            sessionTask = session.uploadTask(with: request, from: requestData) { [callbackQueue] data, response, error in
+            sessionTask = session.uploadTask(with: request, from: requestData) { [callbackQueue, configuration] data, response, error in
                 callbackQueue.async {
-                    if error != nil {
-                        completion(.failure(error!))
+                    if let error = error {
+                        let nsError = error as NSError
+                        if nsError.domain == NSURLErrorDomain {
+                            switch nsError.code {
+                            case NSURLErrorCannotConnectToHost, NSURLErrorNetworkConnectionLost, NSURLErrorCannotFindHost, NSURLErrorAppTransportSecurityRequiresSecureConnection:
+                                configuration.offline = true
```

Comment:
Do we see a need to turn this back online after some time? Or do we want to just try uploading the next time the user comes online?

Reply:
As per the description, we are using NWPathMonitor to monitor connectivity changes and we will automatically reconnect if that detects a new network path is found.

Reply:
ahh interesting. Forgot about the offline checker workflow :)